### PR TITLE
changes done to replace tender icon with pad lock if additional icon …

### PR DESCRIPTION
**Pull request by chandankumarcentric**
https://petcoalm.atlassian.net/browse/PDPOS-9171

Commerce issue #7888

refer to commerce PR https://github.com/JumpMind/commerce/pull/7906

BEFORE:
image

After:
Desktop:
image

Ipad:
image

Mobile:
image

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html [hunk 1]
```diff
@@ -1,12 +1,13 @@
 <button responsive-class #button mat-flat-button [disabled]="disabled" (click)="clickFn()" [type]="inputType"
     color=primary-inverse class="option-item" tabindex="0">
 
-    <app-icon [iconName]="optionIcon" [iconClass]="optionSize" [class.muted-color]="disabled"
+    <app-icon *ngIf="!additionalIcon" [iconName]="optionIcon" [iconClass]="optionSize" [class.muted-color]="disabled"
```
Comment (jrmihalick) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html:4]: Leaving this attribute named as `additionalIcon` is going to be pretty confusing after this change. How about we rename to `iconOverride`?  @jchirico8, @stojankz, @chenson42 ... any opinions?

Comment (jchirico8) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html:4]: Is this change necessary even? If you're replacing the existing icon with a lock can't that be done server side on the tender option itself?

Comment (jrmihalick) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html:4]: @jchirico8 Yes, good point.  @chandankumarcentric : make it so.  😄

Comment (chandankumarcentric) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html]: i did the changes. I need help in handling this part in commerce PR:
https://github.com/JumpMind/commerce/pull/7906/files#diff-dd439fbbe655983250142be3c94021d75d29f735e5397b8780498a8725a1966c

Comment (chandankumarcentric) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html]: or should i keep the logic for additionalIcon and only handle TenderOptions case on server side? Please suggest.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/status/status-details/selector/peripheral-selector.component.html [hunk 2]
```diff
@@ -16,7 +16,6 @@
         appArrowTabItem
         [attr.tabindex]="i"
         [optionTitle]="o.displayName"
-        [additionalIcon]="o.powerLevelIcon"
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/status/status-details/selector/peripheral-selector.component.html:19]: This powerLevelIcon is still required. I think if @JJStarrett used the additionalIcon in its existing location to show the payment device battery level, then you shouldn't change any of this

Comment (chandankumarcentric) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/status/status-details/selector/peripheral-selector.component.html]: > This powerLevelIcon is still required. I think if @JJStarrett used the additionalIcon in its existing location to show the payment device battery level, then you shouldn't change any of this

Thats what i was thinking to leave the additionalIcon as it is and handle only for TenderOption. Done please check now.

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html (hunk 1 above)
Comment (chandankumarcentric) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/option-button/option-button.component.html]: Done. Thanks.

File: openpos-flow/src/main/java/org/jumpmind/pos/core/ui/ActionItem.java [hunk 3]
```diff
@@ -50,6 +50,8 @@ public class ActionItem implements Serializable {
     protected String imageUrl;
     @Builder.Default
     protected boolean enabled = true;
+    @Builder.Default
+    protected boolean iconMuted = false;
```
Comment (jchirico8) [inline openpos-flow/src/main/java/org/jumpmind/pos/core/ui/ActionItem.java:54]: Primitive booleans default to false automatically. You don't need this or the builder.default
```suggestion
    protected boolean iconMuted;
```

Comment (chandankumarcentric) [reply openpos-flow/src/main/java/org/jumpmind/pos/core/ui/ActionItem.java]: right. Thanks.

**Review by jrmihalick — APPROVED** [review]
Don't forget we need this in 3.3 also.

**Pull request merged**